client/engine/engine.go

```diff
@@ -721,3 +727,22 @@ func (e *Engine) recordMessageMetrics(message protocols.Message) {
 	e.metrics.RecordQueueLength(fmt.Sprintf("msg_payload_size,sender=%s,receiver=%s", e.store.GetAddress(), message.To), totalPayloadsSize)
 	e.metrics.RecordQueueLength(fmt.Sprintf("msg_size,sender=%s,receiver=%s", e.store.GetAddress(), message.To), len(raw))
 }
+
+func (e *Engine) checkError(err error) {
+	if err != nil {
+
+		e.logger.Err(err).Msgf("%s, error in run loop", e.store.GetAddress())
+
+		for _, nonFatalError := range nonFatalErrors {
```

Comment:
Would it make sense to invert this logic, and instead of keeping a list of nonFatalErrors, keep a list of fatalErrors and check against that? I think in production we want the default to be non-fatal. So far we have (I think) many examples of non fatal errors, but not so many (none?) fatal error.

Reply:
I am open to either approach. My thinking was that for now we could still lean toward panicking to highlight unexpected errors

Reply:
Sure, OK. Seems safer this way.